src/cogent3/core/new_alignment.py

```diff
@@ -3096,174 +3315,881 @@ def gapped_bytes_value(self) -> bytes:
+        annotation_db if present.
+        """
+        return self.__class__(
+            seqs_data=self._seqs_data,
+            slice_record=self._slice_record[::-1],
+            names=self.names,
+            info=self.info,
+            moltype=self.moltype,
+            source=self.source,
+        )
+
+    def alignment_quality(self, app_name: str = "ic_score", **kwargs):
+        """
+        Computes the alignment quality using the indicated app
+
+        Parameters
+        ----------
+        app_name
+            name of an alignment score calculating app, e.g. 'ic_score',
+            'cogent3_score', 'sp_score'
+
+        kwargs
+            keyword arguments to be passed to the app. Use
+            ``cogent3.app_help(app_name)`` to see the available options.
+
+        Returns
+        -------
+        float or a NotCompleted instance if the score could not be computed
+        """
+        app = get_app(app_name, **kwargs)
+        return app(self)
+
+    def iter_positions(
+        self, pos_order: list = None
+    ) -> typing.Iterator[list, list, list]:
+        """Iterates over positions in the alignment, in order.
+
+        Parameters
+        ----------
+        pos_order
+            list of indices specifying the column order. If None, the
+            positions are iterated in order.
+
+        Returns
+        -------
+        yields lists of elemenets for each position (column) in the alignment
+        """
+        # refactor: array
+        # this could also iter columns of indices as a numpy array - could be an optional arg
+        pos_order = pos_order or range(self._seqs_data.align_len)
+        for pos in pos_order:
+            yield [str(self[seq][pos]) for seq in self.names]
+
+    positions = property(iter_positions)
+
+    def get_position_indices(
+        self, f: Callable, native: bool = False, negate: bool = False
+    ) -> list[int]:
+        """Returns list of column indices for which f(col) is True.
+
+        f : callable
+          function that returns true/false given an alignment position
+        native : boolean
+          if True, and ArrayAlignment, f is provided with slice of array
+          otherwise the string is used
+        negate : boolean
+          if True, not f() is used
+        """
+        # refactor:
+        # type hint for f
+        # implement native
+        if negate:
+
+            def new_f(x):
+                return not f(x)
+
+        else:
+            new_f = f
+
+        result = [i for i, col in enumerate(self.positions) if new_f(col)]
+
+        return result
+
+    def get_gap_array(self, include_ambiguity: bool = True) -> numpy.ndarray:
+        """returns bool array with gap state True, False otherwise
+
+        Parameters
+        ----------
+        include_ambiguity
+            if True, ambiguity characters that include the gap state are
+            included
+        """
+        result = numpy.full((self.num_seqs, len(self)), False, dtype=bool)
```

Comment:
Add a comment about redesign. The method for returning a gap array should be on the IndelMap class.

Reply:
This would only work for the case where `include_ambiguity` is False (it defaults to True), otherwise `IndelMap` would need to know about the seq and alphabet